test/forward/forward.jl

```diff
@@ -1,13 +1,22 @@
-using Zygote, Test
+using ChainRulesCore
+using Test
+using Zygote
 using NNlib: relu
 
-D(f, x) = pushforward(f, x)(1)
+foo(x) = x
+
+# This intentionally has the wrong definition so that we can detect if Zygote is using it.
+ChainRulesCore.frule((_, dx), ::typeof(foo), x) = x, 2 * dx
+
+D(f, x...) = pushforward(f, x...)(1)
+
+@test D(foo, 1.0) == 2
 
 @test D(x -> sin(cos(x)), 0.5) == -cos(cos(0.5))*sin(0.5)
 
 @test D(x -> D(cos, x), 0.5) == -cos(0.5)
```

Comment:
My implementation at the minute doesn't like nested AD. There's a bunch of debugging still to do, so it's entirely possible that it's innocuous. In any case, @oxinabox or @DhairyaLGandhi if you get a minute, could you take a look a the error that this line generates?